provides.py

```diff
@@ -10,3 +10,54 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+
+import charmhelpers.contrib.network.ip as ch_net_ip
+
+from charms.reactive import Endpoint
+from charms.reactive import set_flag, clear_flag
+from charms.reactive import not_unless, when, when_any
+
+
+class MySQLMonitor(Endpoint):
+    @property
+    def relation_ip(self):
+        """Get IP of related endpoint.
+
+        :return: IP address of related endpoint
+        :rtype: Optional[str]
+        """
+        return ch_net_ip.get_relation_ip(self.endpoint_name)
+
+    @when("endpoint.{endpoint_name}.joined")
+    def joined(self):
+        """Handle joined relation."""
+        set_flag(self.expand_name("{endpoint_name}.connected"))
+
+    @when_any("endpoint.{endpoint_name}.broken",
+              "endpoint.{endpoint_name}.departed")
+    def broken_or_departed(self):
+        """Handle broken relation."""
+        for relation in self.relations:
+            relation.to_publish_raw["host"] = None
```

Comment:
One question is whether you want to use `to_publish_raw` or `to_publish`, which depends on the format of the data back and forth. Here it looks as though the intention is just base key/value pairs and I think that's okay, however the preferred approach is to use the `to_publish` format. (reference https://charmsreactive.readthedocs.io/en/latest/charms.reactive.relations.html?highlight=to_publish#charms.reactive.endpoints.Relation.to_publish)

nit: this might be a bit cleaner as:

```python
relation.to_publish_raw.update({
    'host': None,
    'port': None,
    'username': None,
    'password': None
})
```

Reply:
Thanks for the tip (I didn't know about it), I'll definitely use the `to_publish` function, as it makes more sense.